client/components/sites-add-new-site/test/index.tsx

```diff
@@ -0,0 +1,58 @@
+/**
+ * @jest-environment jsdom
+ */
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { render, screen, fireEvent } from '@testing-library/react';
+import React from 'react';
+import { SitesAddNewSitePopover } from '../index';
+
+// Mock dependencies
+jest.mock( '@automattic/calypso-analytics', () => ( {
+	recordTracksEvent: jest.fn(),
+} ) );
+
+jest.mock( 'i18n-calypso', () => ( {
+	useTranslate: () => ( text: string ) => text,
+	localize: ( x ) => x,
+} ) );
+
+// Mock the AsyncContent component to avoid loading issues in tests
+jest.mock( '../content/async', () => ( {
+	AsyncContent: () => <div data-testid="async-content">Async Content</div>,
+} ) );
+
+describe( 'SitesAddNewSitePopover', () => {
+	beforeEach( () => {
+		jest.clearAllMocks();
+	} );
+
+	it( 'renders with full label when not in compact mode', () => {
+		render( <SitesAddNewSitePopover showCompact={ false } /> );
+		expect( screen.getByText( 'Add new site' ) ).toBeInTheDocument();
+	} );
+
+	it( 'renders without label in compact mode', () => {
+		render( <SitesAddNewSitePopover showCompact /> );
+		expect( screen.queryByText( 'Add new site' ) ).not.toBeInTheDocument();
+	} );
+
+	it( 'tracks event when dropdown is opened', () => {
+		render( <SitesAddNewSitePopover showCompact={ false } /> );
+		const button = screen.getByRole( 'button' );
+		fireEvent.click( button );
+		expect( recordTracksEvent ).toHaveBeenCalledWith(
+			'calypso_sites_dashboard_new_site_action_click_open'
+		);
+	} );
+
+	it( 'closes popover when clicking the button again', () => {
+		render( <SitesAddNewSitePopover showCompact={ false } /> );
+		const button = screen.getByRole( 'button' );
+		fireEvent.click( button );
+		expect( screen.getByTestId( 'async-content' ) ).toBeInTheDocument();
+
+		// Click button again
+		fireEvent.click( button );
+		expect( screen.queryByTestId( 'async-content' ) ).not.toBeInTheDocument();
+	} );
+} );
```

Comment:
Might it be worth adding tests case to verify that the popover is hidden when clicking outside the popover area and/or hitting ESC with the popover in focus? Both are technically a test of the underlying component, but they are both key parts of the behaviour we want, and would make it easier to verify the behaviour if we switch the underlying component in future.

Reply:
I would prefer not to test the underlying component's blur/keydown events here. Test environments vary, and we can't reliably predict where the event is bound. You would need to have some understanding of how the underlying component works to trigger the right event. Triggering events on the body won't work in this case.